lib/matplotlib/text.py

```diff
@@ -158,6 +157,7 @@ def __init__(self,
         self._linespacing = linespacing
         self.set_rotation_mode(rotation_mode)
         self.update(kwargs)
+        self._cached = cbook.maxdict(50)
```

Comment:
now that we are caching on the instance rather than the class, I think that this will now be bounded to only 1 or 2 entries?  If so we are never going to hit the max of 50 and can get away with using a simple `dict` here.